src/main.ts

```diff
@@ -16,48 +21,101 @@ async function commandExists(cmd: string): Promise<boolean> {
+  if (!sbtExists) {
+    core.setFailed('Not found sbt command')
+    return
+  }
+
+  const input = {
+    projects: core
+      .getInput('projects')
+      .split(' ')
+      .filter(value => value.length > 0),
+    scalaVersions: core
+      .getInput('scala-versions')
+      .split(' ')
+      .filter(value => value.length > 0),
+  }
+
+  await cli.exec('sbt', [`githubSubmitDependencyGraph ${JSON.stringify(input)}`], {
+    cwd: baseDir,
+  })
+}
+
+function getMillPath(baseDir: string): string {
+  const millPath = path.join(baseDir, 'mill')
+  const millWPath = path.join(baseDir, 'millw')
+  if (fs.existsSync(millPath)) {
+    return './mill'
+  } else if (fs.existsSync(millWPath)) {
+    return './millw'
+  } else {
+    core.info('Installing mill...')
+    cli.exec('curl', [
+      '-sLo',
+      millPath,
+      `https://github.com/com-lihaoyi/mill/releases/tag/${defaultMillVersion}/${defaultMillVersion}`,
+    ])
+    cli.exec('chmod', ['+x', millPath])
+    return './mill'
+  }
+}
+
+async function runMill(baseDir: string, pluginVersionInput: string): Promise<void> {
+  const millCommand = getMillPath(baseDir)
+
+  const pluginVersion =
+    pluginVersionInput.length === 0 ? defaultMillPluginVersion : pluginVersionInput
+
+  await cli.exec(
+    millCommand,
+    [
+      '--import',
+      `ivy:io.chris-kipp::mill-github-dependency-graph::${pluginVersion}`,
+      'io.kipp.mill.github.dependency.graph.Graph/submit',
+    ],
+    {
+      cwd: baseDir,
+    },
+  )
+}
+
 async function run(): Promise<void> {
   try {
     const token = core.getInput('token')
     core.setSecret(token)
+    process.env['GITHUB_TOKEN'] = token
 
     const baseDirInput = core.getInput('base-dir')
     const baseDir = baseDirInput.length === 0 ? '.' : baseDirInput
-    const projectDir = path.join(baseDir, 'project')
-    if (!fs.existsSync(projectDir)) {
-      core.setFailed(`${baseDir} is not a valid sbt project: missing folder '${projectDir}'.`)
-      return
-    }
 
-    const uuid = crypto.randomUUID()
-    const pluginFile = path.join(projectDir, `github-dependency-graph-${uuid}.sbt`)
-
-    const pluginVersionInput = core.getInput('sbt-plugin-version')
-    const pluginVersion =
-      pluginVersionInput.length === 0 ? defaultPluginVersion : pluginVersionInput
-    const pluginDep = `addSbtPlugin("ch.epfl.scala" % "sbt-github-dependency-graph" % "${pluginVersion}")`
-    await fsPromises.writeFile(pluginFile, pluginDep)
-    const sbtExists = await commandExists('sbt')
-    if (!sbtExists) {
-      core.setFailed('Not found sbt command')
-      return
-    }
+    const pluginVersionInput = core.getInput('plugin-version')
 
-    const input = {
-      projects: core
-        .getInput('projects')
-        .split(' ')
-        .filter(value => value.length > 0),
-      scalaVersions: core
-        .getInput('scala-versions')
-        .split(' ')
-        .filter(value => value.length > 0),
+    if (fs.existsSync(path.join(baseDir, 'build.sc'))) {
```

Comment:
It is very nice to guess for the user and it will work in 99% of the cases. But I suggest we add an optional `build-tool` input to deal with the non-working cases such as:
- there is a `build.sc` and a `build.sbt` file in the directory
- the name of the sbt file is not `build.sbt` (this is allowed by sbt)

Reply:
So my typescript suck and I got pretty annoyed with this. I just used an array of choices instead of an enum since I couldn't find a nice way to do a check without casting to any, and we have that turned off.